Common/Chairloader/ConfigNode.h

```diff
@@ -0,0 +1,247 @@
+
+    ConfigNode(pugi::xml_node node)
+    : m_node(node)
+    {
+        if(m_node){
+            m_name = m_node.name();
+            m_displayName = m_node.attribute("displayName").as_string();
+            m_description = m_node.attribute("description").as_string();
+            m_type = m_node.attribute("type").as_string();
+            m_nodeType = toNodeType(m_type);
+        }
+        else {
+            m_name = "";
+            m_displayName = "";
+            m_description = "";
+            m_type = "";
+            m_nodeType = NodeType::XmlNode;
+        }
+    }
+
+    ConfigNode() : m_node() {
+        m_name = "";
+        m_displayName = "";
+        m_description = "";
+        m_type = "";
+        m_nodeType = NodeType::XmlNode;
+    }
+
+    ConfigNode operator[](const char* name) {
+        return ConfigNode(m_node.child(name));
+    }
+
+    ConfigNode get(const char* name) {
+        return ConfigNode(m_node.child(name));
+    }
+
+    ConfigNode create(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            throw std::runtime_error("Node already exists");
+        }
+        auto node = m_node.append_child(name);
+        node.append_attribute("type") = toString(type).c_str();
+        node.append_child(pugi::node_pcdata).set_value("");
+        return ConfigNode(node);
+    }
+
+    ConfigNode getOrCreate(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            return ConfigNode(m_node.child(name));
+        }
+        return create(name, type);
+    }
+
+    bool exists() const {
+        return m_node;
+    }
+
+    NodeType type() const {
+        return m_nodeType;
+    }
+
+    std::string name() const {
+        return m_name;
+    }
+
+    std::string displayName() const {
+        return m_displayName;
+    }
+
+    std::string description() const {
+        return m_description;
+    }
+
+    pugi::xml_node raw() const {
+        return m_node;
+    }
+
+    template<typename T>
+    T as() const {
+        if(!m_node) {
+            throw std::runtime_error("Node does not exist");
+        }
+        if(m_nodeType == NodeType::Enum){
+            return ConfigNode(m_node.child("selected")).as<T>();
+        } else {
+            return getValue<T>(m_node);
+        }
+    }
+
+    /**
+     * Get the value of the node with a default value if the node is not found.
+     * @return The value of the node or the default value if the node does not exist or is not of the expected type.
+     */
+    template<typename T>
+    T asOr(T defaultValue) const noexcept{
+        try {
+            if(m_nodeType == NodeType::Enum){
+                return ConfigNode(m_node.child("selected")).asOr<T>(defaultValue);
+            }
+            return exists() ? getValue<T>(m_node) : defaultValue;
+        } catch (...) {
+            return defaultValue;
+        }
```

Comment:
[Do not use exceptions for control flow.](https://isocpp.org/wiki/faq/exceptions#why-not-exceptions) This may also hide some other unexpected error. The code also partially duplicates code of `as()`